### PR TITLE
Interactive support

### LINKED ISSUE #28: interactive message support
I plan to use this as my bot server but I would need interactive message support.

How would I be able to support that? If you don't have time to do it yourself, could you point some way and I will try to add it, it is important for me that I can do interactive messages.

### PR DESCRIPTION
**Pull request by confact**
Tested and work on my local env.

Added:
* update method to update a message directly on slack
* combined methods for interactive respond in remote, server and bot to forward the request data from the webhook

fixes #28

### CODE REVIEW

File: lib/slack_bot_server/bot.rb [hunk 1]
```diff
@@ -100,6 +100,34 @@ def say(options)
...
 
+  # Update a message on Slack
+  # @param options [Hash] a hash containing any of the following:
+  #    channel:: the name ('#general'), or the ID of the channel to send to
+  #    text:: the actual text of the message
+  #    username:: the name the message should appear from; defaults to the
+  #               value given to `username` in the Bot class definition
+  #    icon_url:: the image url to use as the avatar for this message;
+  #               defaults to the value given to `icon_url` in the Bot
+  #               class definition
```
Comment (lazyatom) [inline lib/slack_bot_server/bot.rb]: I believe that `message_ts` is a required option here, and `username` and `icon_url` are not valid options: https://api.slack.com/methods/chat.update

File: lib/slack_bot_server/bot.rb [hunk 2]
```diff
@@ -100,6 +100,34 @@ def say(options)
...
+  def update(options)
+    message = symbolize_keys(options)
+
+    if rtm_incompatible_message?(message)
+      debug "Sending via Web API", message
+      client.web_client.chat_postMessage(message)
+    else
+      debug "Sending via RTM API", message
+      client.message(message)
```
Comment (lazyatom) [inline lib/slack_bot_server/bot.rb]: Is it actually possible to update a message using the RTM API? I can't see anything in the documentation that suggests it is, but perhaps I missed something?

File: lib/slack_bot_server/server.rb [hunk 3]
```diff
@@ -59,7 +59,7 @@ class SlackBotServer::Server
   def initialize(queue: SlackBotServer::LocalQueue.new)
     @queue = queue
     @bots = {}
-    @add_proc = -> (token) { SlackBotServer::SimpleBot.new(token: token) }
+    @add_proc = ->(token) { SlackBotServer::SimpleBot.new(token: token) }
```
Comment (lazyatom) [inline lib/slack_bot_server/server.rb]: I'd prefer it if these formatting changes were in a separate pull request if you think they're important.

**Review by lazyatom — CHANGES_REQUESTED** [review]
OK, I think I understand the general approach you're hoping for.

The way I've handled updating messages in my own app is just by returning the message JSON from the controller action, but I can see the desire to keep at least that JSON-creation code close to the bot, so adding an `update` method is good and I'd support that.

However, I don't see much benefit from defining this `interactive_respond` method, when you can already send the `call` message to call arbitrary methods on the bot instance (including one called `handle_interactive`). If you prefer that everything is mediated through a separate method, you can still define that on your bot subclass and then in your app write a wrapper method that translates `remote.handle_interactive(*args)` to `remote.call(:handle_interactive, *args)`

I'm happy to entertain code formatting changes, but I'd prefer to do that separately from feature changes, and consistently across the whole codebase. Switching to single quotes is fine, but let's do it everywhere we can, if we're doing it.

Lastly, but importantly, we need tests! :smile:

**Comment by confact** [comment]
Thanks for your feedback. 

I have no excuse for missing tests. I will fix that.
I have totally missed the call method so I will use that instead, my bad.

File: lib/slack_bot_server/bot.rb (hunk 1 above)
Comment (confact) [reply lib/slack_bot_server/bot.rb]: This is fixed now, please check again.

File: lib/slack_bot_server/bot.rb (hunk 2 above)
Comment (confact) [reply lib/slack_bot_server/bot.rb]: This is fixed now, please check again.

File: lib/slack_bot_server/server.rb (hunk 3 above)
Comment (confact) [reply lib/slack_bot_server/server.rb]: This is fixed now, please check again.

**Comment by lazyatom** [comment]
Apologies for the delay with this; I've merged a cleaned up version of commits here: https://github.com/exciting-io/slack-bot-server/commit/f8c68ff98dbce82a69a80b85e7fe72378ece81d6

**Pull request closed without merging**